LibreNMS/OS/Vrp.php

```diff
@@ -755,12 +755,14 @@ public function discoverVlanPorts(Collection $vlans): Collection
                         $vlansOnPort = StringHelpers::bitsToIndices($vlanArray[$oid]);
                         foreach ($vlansOnPort as $vlanIdOnPort) {
                             $vlanIdOnPort = ($hilo == 'High') ? ($vlanIdOnPort + 2047) : ($vlanIdOnPort - 1);
-                            $ports->push(new PortVlan([
-                                'vlan' => $vlanIdOnPort,
-                                'baseport' => $baseport,
-                                'untagged' => 0,
-                                'port_id' => PortCache::getIdFromIfIndex($portsIndexes[$baseport] ?? 0, $this->getDeviceId()) ?? 0, // ifIndex from device
-                            ]));
+                            if ($vlans->firstWhere('vlan_vlan', $vlanIdOnPort) !== null) {
```

Comment:
```suggestion
                            if ($vlans->contains('vlan_vlan', $vlanIdOnPort)) {
```

this is a bit more obvious what it is doing.